### PR TITLE
Display future product tier

### PR DESCRIPTION
**Pull request by andresilva-guardian**
### Current situation/background

Currently, when a user has scheduled a future change to their subscription (such as downgrading from a higher tier membership to Contributor), this information is not displayed in the Account Overview page. Additionally, the system is incorrectly allowing users to navigate to the Membership Switch flow even when they already have a future plan change scheduled.

### What does this PR change?

This PR makes several improvements to how we handle and display future subscription plans:

1. Added a `futureProductTitle` variable in ProductCard that determines if there's a future plan change scheduled
2. Modified the product details section to display the "Next product" information when a future plan change exists
3. Updated the MembershipSwitch component to redirect users to the Overview page if they already have a future plan change to Contributor tier
4. Fixed an issue in NextPaymentDetails where we were displaying `paymentValue` instead of the current price
5. Added the tier property to the SubscriptionPlan interface to properly support future plan changes

#### Visual Changes
| Before | After |
|--------|--------|
| Screenshot 2025-05-09 at 15 19 56 | Screenshot 2025-05-09 at 15 20 28 |
* NOTE: "Next product" changed to "Switching to" by indication of @Alexguild 

### Next steps/further info

- We may want to add more comprehensive information about planned subscription changes, including when the change will take effect
- Consider adding visual indicators or badges to make future plan changes more noticeable
- We should test this thoroughly across different subscription types to ensure all edge cases are handled correctly
- Documentation should be updated to reflect these changes for the support team

### CODE REVIEW

File: client/components/mma/accountoverview/ProductCard.tsx [hunk 1]
```diff
@@ -431,6 +440,12 @@ export const ProductCard = ({
 											</dd>
 										</div>
 									)}
+								{futureProductTitle && (
+									<div>
+										<dt>Switching to</dt>
+										<dd>{futureProductTitle}</dd>
```
Comment (johnduffell) [inline client/components/mma/accountoverview/ProductCard.tsx:446]: just to understand better - if they are switching from 95 to 49 EUR is the 49 EUR (only) going to show in the the next payment amount field?
I'm not sure if there's any situation where they would have a forward dated switch to a discounted plan, so it would be ok at this point, but it does seem like the charges can be complex.

Another option (not for this PR I'm sure) would be to show a billing preview of the next few payments.

File: client/components/mma/shared/NextPaymentDetails.tsx [hunk 2]
```diff
@@ -89,6 +90,9 @@ export const getNextPaymentDetails = (
 			isNewPaymentValue,
 			nextPaymentDateKey: 'Next payment date',
 			nextPaymentDateValue,
+			currentPriceValue: `${mainPlan.currency}${(
+				(subscription.plan?.price ?? 0) / 100
```
Comment (johnduffell) [inline client/components/mma/shared/NextPaymentDetails.tsx:94]: I saw a comment in a PR saying that subscription.plan is deprecated and we should be using current/futurePlans fields .  Not sure if they have the right information to switch over (or how aspirational that comment was)

File: client/components/mma/accountoverview/ProductCard.tsx (hunk 1 above)
Comment (andresilva-guardian) [reply client/components/mma/accountoverview/ProductCard.tsx:446]: Hey @johnduffell , the thing is, currently you can read "you're supporting the Guardian with €49 (...)" - BUT THIS IS NOT TRUE - you paid the 95 euros until that moment to have all the "exclusive extras". I believe is "false advertising" allowing the user to understand that it could get the "exclusive extras" por 49 euros when in fact, for 49 euros there will be no "exclusive extras" anymore (next year). Makes sense?

File: client/components/mma/cancel/cancellationSaves/membership/MembershipSwitch.tsx [hunk 3]
```diff
@@ -262,6 +261,17 @@ export const MembershipSwitch = () => {
...
 		}
 	};
+
+	if (
+		membership.subscription.futurePlans.length > 0 &&
+		membership.subscription.futurePlans[0].tier &&
+		membership.subscription.futurePlans[0].tier === 'Contributor'
+	) {
+		// Switch already planned, so navigate to Overview
+		navigate('/');
```
Comment (johnduffell) [inline client/components/mma/cancel/cancellationSaves/membership/MembershipSwitch.tsx:271]: is this just a fall back if they navigate directly to the page?  (Even if not, I'm sure it's better than letting them switch when they shouldn't.)

File: client/components/mma/accountoverview/ProductCard.tsx [hunk 4]
```diff
@@ -204,6 +204,15 @@ export const ProductCard = ({
 		productDetail.subscription.nextPaymentDate !==
 			productDetail.subscription.potentialCancellationDate;
 
+	const futureProductTitle =
+		productDetail.subscription.futurePlans.length > 0 &&
+		productDetail.subscription.futurePlans[0].tier &&
```
Comment (johnduffell) [inline client/components/mma/accountoverview/ProductCard.tsx:209]: nit pick-just wondering if these two lines could be combined with some kind of question mark, I'm not sure the exact syntax but this kind of thing
```suggestion
		productDetail.subscription.futurePlans[0]?.tier &&
```

File: client/components/mma/accountoverview/ProductCard.tsx (hunk 1 above)
Comment (johnduffell) [reply client/components/mma/accountoverview/ProductCard.tsx:445]: yeah got it, so when we say "you're supporting..with..." then it has to be the current charge, and actually it has to be clear when they pay the next amount it's for something else.
And actually I rechecked your screenshots and it looks pretty clear with the change of product and amount represented together. 👍

**Review by johnduffell — APPROVED** [review]
looks good, only a few questions/nit picks.
I noticed you said the documentation should be updated, but I didn't know which documentation that is.  I also wanted to mention that if you feel like it could cause customer confusion we can also contact the CSR and userhelp to give them a heads up.

File: client/components/mma/shared/NextPaymentDetails.tsx (hunk 2 above)
Comment (johnduffell) [reply client/components/mma/shared/NextPaymentDetails.tsx:94]: this PR is the one that says plan is deprecated https://github.com/guardian/members-data-api/pull/371

File: client/components/mma/cancel/cancellationSaves/membership/MembershipSwitch.tsx (hunk 3 above)
Comment (andresilva-guardian) [reply client/components/mma/cancel/cancellationSaves/membership/MembershipSwitch.tsx:271]: @johnduffell Just to let you know, I have added a change to remove the button that redirects to the membership switch page if the future plan is to be a Contributor.

File: client/components/mma/shared/NextPaymentDetails.tsx (hunk 2 above)
Comment (andresilva-guardian) [reply client/components/mma/shared/NextPaymentDetails.tsx:94]: @johnduffell I will add this to the code
Screenshot 2025-05-12 at 12 14 26

So IDEs can show this variable formatted with Strikethrough
Screenshot 2025-05-12 at 12 16 40

I will of course fix it right now also 👍

Comment (andresilva-guardian) [reply client/components/mma/shared/NextPaymentDetails.tsx]: @johnduffell Code fixed ✅
